pyomo/core/plugins/transform/deactivate_trivial_constraints.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: UTF-8 -*-
+"""Transformation to deactivate trivial constraints."""
+import textwrap
+from math import fabs
+
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.kernel.component_set import ComponentSet
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+
+__author__ = "Qi Chen <https://github.com/qtothec>"
+
+
+class TrivialConstraintDeactivator(IsomorphicTransformation):
+    """Deactivates trivial constraints of form constant = constant."""
+
+    alias('core.deactivate_trivial_constraints',
+          doc=textwrap.fill(textwrap.dedent(__doc__.strip())))
+
+    def __init__(self):
+        """Initialize the transformation."""
+        super(TrivialConstraintDeactivator, self).__init__()
+        self.tolerance = 1E-13
+
+    def _apply_to(self, instance, tmp=False, ignore_infeasible=False):
+        """Apply the transformation."""
+        if tmp and not hasattr(instance, '_tmp_trivial_deactivated_constrs'):
+            instance._tmp_trivial_deactivated_constrs = ComponentSet()
```

Comment:
What is the design goal/requirement for including local data about deactivated components?  When would a user want to revert?  This logic isn't generally supported in Pyomo, so why do this for this transformation and not others?

Reply:
A constraint may become trivial as a result of some variables (e.g. discrete ones) being fixed. In some cases, we do not want to send these constraints to an NLP solver (like IPOPT), because they will return nonsensical duals that foul up cutting plane generation, as @bernalde found. However, we may want to restore these constraints later when the variables are unfixed.